platform/default/src/mbgl/storage/database_file_source.cpp

```diff
@@ -27,7 +27,7 @@ class DatabaseFileSourceThread {
             offlineResponse->noContent = true;
             offlineResponse->error =
                 std::make_unique<Response::Error>(Response::Error::Reason::NotFound, "Not found in offline database");
-        } else if (!offlineResponse->isUsable()) {
+        } else if (resource.storagePolicy == Resource::StoragePolicy::Volatile || !offlineResponse->isUsable()) {
```

Comment:
`resource.storagePolicy == Resource::StoragePolicy::Volatile` check should be done before we even go to the [database](https://github.com/mapbox/mapbox-gl-native/pull/16422/files#diff-624e23e19f1608acc747a18f380e94f1L24). Also, not sure why such request even made from database for `Resource::StoragePolicy::Volatile` resource? Should we also short-circuit it in main resource loader?

Reply:
I made it in order to distinguish between error messages, but skipping db request is probably more useful